misk/src/main/kotlin/misk/security/authz/AccessAnnotation.kt

```diff
@@ -0,0 +1,45 @@
+package misk.security.authz
+
+import kotlin.reflect.KClass
+
+/**
+ * Use this to alias an annotation to a set of services and roles. This can be used to decouple code
+ * that needs access control from the policy that defines it.
+ *
+ * To demonstrate, let's define an access annotation. By convention these annotations are suffixed
+ * `Access`:
+ *
+ * ```
+ * @Retention(AnnotationRetention.RUNTIME)
+ * @Target(AnnotationTarget.FUNCTION)
+ * annotation class PaleontologistAccess
+ * ```
+ *
+ * Next we define actions that apply our annotation:
+ *
+ * ```
+ * class DiscoverDinosaurAction : WebAction {
+ *   @Get("/discover")
+ *   @PaleontologistAccess
+ *   fun discover()
+ * }
+ *
+ * class DigUpDinosaurAction : WebAction {
+ *   @Get("/dig")
+ *   @PaleontologistAccess
+ *   fun dig()
+ * }
+ * ```
+ *
+ * Finally we use multibindings to specify which services and roles are permitted:
+ *
+ * ```
+ * multibind<AccessAnnotation>().toInstance(AccessAnnotation(
+ *     DigUpDinosaurAction::class, roles = listOf("paleontologist", "intern")))
```

Comment:
Should this be ``PaleontologistAccess``?

Reply:
Yes!